packages/react/src/components/Menu/Menu.tsx

```diff
@@ -225,6 +225,13 @@ const Menu = forwardRef<HTMLUListElement, MenuProps>(function Menu(
   }
 
   function focusItem(e?: React.KeyboardEvent<HTMLUListElement>) {
+    const items = focusableItems;
+
+    if (!items || items.length === 0) return;
+
+    const validItems = items.filter((item) => item?.ref?.current);
+    if (validItems.length === 0) return;
```

Comment:
Just a thought, `focusableItems` can be used directly right?

```suggestion
    const validItems = focusableItems?.filter(item => item?.ref?.current);
    if (!validItems?.length) return;
```